### ISSUE TITLE
pdfsizeopt creates broken output PDF for some inputs

### ISSUE THREAD

**Issue by hannob** [pts/pdfsizeopt#69]
I'll attach a zip file containing two pdf files. One is a reduced example of a document I have here where pdfsizeopt creates no valid output.
The other is the output, created by the latest version of pdfsizeopt installed according to the instructions for linux (2017-01-24). evince and chrome both report it as broken.

I have an older version of pdfsizeopt here that I've been using for a while and that will produce a pdf output that can be read, but breaks the font in the diagram.

[examplefiles.zip](https://github.com/pts/pdfsizeopt/files/1545910/examplefiles.zip)

**Comment by davdittrich** [comment]
I observe the same problem with the current version (b243e25). 
cpdf gives me, e.g., the following error message:
```
Recovering from Lex error: Pdf.PDFError("Failure lexing stream dict whilst reading file bytes at position 43236")
Attempting to reconstruct the malformed pdf 1.pdf...
Read 1687 objects
cpdf encountered an error. Technical details follow:

Failed to read PDF - initial error was
Pdf.PDFError("Could not read object whilst reading file bytes at position 43822")

final error was 
Pdf.PDFError("No /Root entry whilst reading file 1.pdf at position 9274404")

```

Reverting to 84c2d80 allows to optimize pdfs so that they seems to be fine.

**Comment by pts** [comment]
Thank you for reporting this! Indeed the newest pdfsizeopt creates *brokenoutput.pdf* and Google Chrome is not able to open it. Evince is not able to open it either. It fails with the following error:

```Syntax Error: Couldn't find trailer dictionary
Syntax Error: Catalog object is wrong type (cmd)
Syntax Error: Couldn't find trailer dictionary
Syntax Error: Couldn't find trailer dictionary
Syntax Error: Catalog object is wrong type (null)
Syntax Error: Couldn't read page catalog
Syntax Error: Couldn't find trailer dictionary
Syntax Error: Catalog object is wrong type (cmd)
Syntax Error: Couldn't find trailer dictionary
Syntax Error: Couldn't find trailer dictionary
Syntax Error: Catalog object is wrong type (null)
Syntax Error: Couldn't read page catalog```

Investigating...

**Comment by pts** [comment]
Fixed in 5649cdd1d362fd8a5225fd3545818cabab0f6e69.

Thank you again for reporting this bug. Please upgrade to the latest pdfsizeopt, it should work as intended.
